MediaBrowser.Model/Dlna/DeviceProfile.cs

```diff
@@ -1,506 +1,517 @@
-        /// </summary>
-        [DefaultValue(0)]
-        public int TimelineOffsetSeconds { get; set; }
-
-        /// <summary>
-        /// Gets or sets a value indicating whether RequiresPlainVideoItems.
-        /// </summary>
-        [DefaultValue(false)]
-        public bool RequiresPlainVideoItems { get; set; }
-
-        /// <summary>
-        /// Gets or sets a value indicating whether RequiresPlainFolders.
-        /// </summary>
-        [DefaultValue(false)]
-        public bool RequiresPlainFolders { get; set; }
-
-        /// <summary>
-        /// Gets or sets a value indicating whether EnableMSMediaReceiverRegistrar.
-        /// </summary>
-        [DefaultValue(false)]
-        public bool EnableMSMediaReceiverRegistrar { get; set; }
-
-        /// <summary>
-        /// Gets or sets a value indicating whether IgnoreTranscodeByteRangeRequests.
-        /// </summary>
-        [DefaultValue(false)]
-        public bool IgnoreTranscodeByteRangeRequests { get; set; }
-
-        /// <summary>
-        /// Gets or sets the XmlRootAttributes.
-        /// </summary>
-        public XmlAttribute[] XmlRootAttributes { get; set; } = Array.Empty<XmlAttribute>();
-
-        /// <summary>
-        /// Gets or sets the direct play profiles.
-        /// </summary>
-        public DirectPlayProfile[] DirectPlayProfiles { get; set; } = Array.Empty<DirectPlayProfile>();
-
-        /// <summary>
-        /// Gets or sets the transcoding profiles.
-        /// </summary>
-        public TranscodingProfile[] TranscodingProfiles { get; set; } = Array.Empty<TranscodingProfile>();
-
-        /// <summary>
-        /// Gets or sets the container profiles.
-        /// </summary>
-        public ContainerProfile[] ContainerProfiles { get; set; } = Array.Empty<ContainerProfile>();
-
-        /// <summary>
-        /// Gets or sets the codec profiles.
-        /// </summary>
-        public CodecProfile[] CodecProfiles { get; set; } = Array.Empty<CodecProfile>();
-
-        /// <summary>
-        /// Gets or sets the ResponseProfiles.
-        /// </summary>
-        public ResponseProfile[] ResponseProfiles { get; set; } = Array.Empty<ResponseProfile>();
-
-        /// <summary>
-        /// Gets or sets the subtitle profiles.
-        /// </summary>
-        public SubtitleProfile[] SubtitleProfiles { get; set; } = Array.Empty<SubtitleProfile>();
-
-        /// <summary>
-        /// The GetSupportedMediaTypes.
-        /// </summary>
-        /// <returns>The .</returns>
-        public string[] GetSupportedMediaTypes()
+        get => _supportedMediaTypeString;
+        set
         {
-            return ContainerProfile.SplitValue(SupportedMediaTypes);
+            _supportedMediaTypeString = value ?? AllMedia;
+            var test = _supportedMediaTypeString
+                .Split(',')
+                .SelectMany(i => Enum.TryParse<DlnaProfileType>(i, true, out var result) && Enum.IsDefined(typeof(DlnaProfileType), result)
```

Comment:
Can IsDefined return false when TryParse returns true?

Reply:
I don't think so. Should I remove it?